### ISSUE TITLE
vkconfig: becoms unresponsive when creating a new layer configuration

### ISSUE THREAD

**Issue by johnzupin** [LunarG/VulkanTools#1446]
Environment:
- Ubuntu 20.04
- Nvidia

Steps to reproduce:
1. Right click and select new
2. Give name and description
3. Select Overridden/force for the monitor and apidump layer. 
4. Hit ok, window disappears and app freezes - Have to kill and restart the app

**Comment by johnzupin** [comment]
Still seeing a problem as of: v1.2.170-261-g5aeab5532 

It looks like an assert is being thrown now.

```
Starting program: /home/lunarg/VulkanTools/dbuild/vkconfig/vkconfig 
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
[New Thread 0x7ffff2d9f700 (LWP 4525)]
[New Thread 0x7ffff1204700 (LWP 4526)]
[New Thread 0x7ffff0a03700 (LWP 4527)]
[New Thread 0x7fffe3efe700 (LWP 4528)]
UNASSIGNED-khronos-validation-createinstance-status-message(INFO / SPEC): msgNum: -671457468 - Validation Information: [ UNASSIGNED-khronos-validation-createinstance-status-message ] Object 0: handle = 0x555555b5d920, type = VK_OBJECT_TYPE_INSTANCE; | MessageID = 0xd7fa5f44 | Khronos Validation Layer Active:
    Settings File: Found at /home/lunarg/.local/share/vulkan/settings.d/vk_layer_settings.txt specified by VkConfig application override.
    Current Enables: None.
    Current Disables: VK_VALIDATION_FEATURE_DISABLE_THREAD_SAFETY_EXT.

    Objects: 1
        [0] 0x555555b5d920, type: 1, name: NULL
vkconfig: /home/lunarg/VulkanTools/vkconfig/widget_setting_filesystem.cpp:41: WidgetSettingFilesystem::WidgetSettingFilesystem(QTreeWidget*, QTreeWidgetItem*, const SettingMetaFilesystem&, SettingDataSet&): Assertion `&data' failed.

Thread 1 "vkconfig" received signal SIGABRT, Aborted.
__GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
50	../sysdeps/unix/sysv/linux/raise.c: No such file or directory.
(gdb) bt
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
#1  0x00007ffff6705859 in __GI_abort () at abort.c:79
#2  0x00007ffff6705729 in __assert_fail_base (fmt=0x7ffff689b588 "%s%s%s:%u: %s%sAssertion `%s' failed.\n%n", 
    assertion=0x5555556892db "&data", file=0x555555689290 "/home/lunarg/VulkanTools/vkconfig/widget_setting_filesystem.cpp", line=41, 
    function=<optimized out>) at assert.c:92
#3  0x00007ffff6716f36 in __GI___assert_fail (assertion=0x5555556892db "&data", 
    file=0x555555689290 "/home/lunarg/VulkanTools/vkconfig/widget_setting_filesystem.cpp", line=41, 
    function=0x555555689210 "WidgetSettingFilesystem::WidgetSettingFilesystem(QTreeWidget*, QTreeWidgetItem*, const SettingMetaFilesystem&, SettingDataSet&)") at assert.c:101
#4  0x00005555555c24fb in WidgetSettingFilesystem::WidgetSettingFilesystem (this=0x5555564e1410, tree=0x555555be30a0, 
    item=0x5555564a6510, meta=..., data_set=...) at /home/lunarg/VulkanTools/vkconfig/widget_setting_filesystem.cpp:41
#5  0x00005555555b373a in SettingsTreeManager::BuildTreeItem (this=0x7fffffffdd30, parent=0x5555564dff20, meta_set=..., data_set=..., 
    meta_object=...) at /home/lunarg/VulkanTools/vkconfig/settings_tree.cpp:293
#6  0x00005555555b3dce in SettingsTreeManager::BuildTreeItem (this=0x7fffffffdd30, parent=0x555555e53400, meta_set=..., data_set=..., 
    meta_object=...) at /home/lunarg/VulkanTools/vkconfig/settings_tree.cpp:347
#7  0x00005555555b4084 in SettingsTreeManager::BuildGenericTree (this=0x7fffffffdd30, parent=0x555555e53400, parameter=...)
    at /home/lunarg/VulkanTools/vkconfig/settings_tree.cpp:357
#8  0x00005555555b2716 in SettingsTreeManager::CreateGUI (this=0x7fffffffdd30, build_tree=0x555555be30a0)
    at /home/lunarg/VulkanTools/vkconfig/settings_tree.cpp:134
#9  0x00005555555a5e1d in MainWindow::NewClicked (this=0x7fffffffdd00) at /home/lunarg/VulkanTools/vkconfig/mainwindow.cpp:781
#10 0x00005555555a912d in MainWindow::eventFilter (this=0x7fffffffdd00, target=0x555555e6fea0, event=0x7fffffffd170)
    at /home/lunarg/VulkanTools/vkconfig/mainwindow.cpp:1243
#11 0x00007ffff6ea564b in QCoreApplicationPrivate::sendThroughObjectEventFilters(QObject*, QEvent*) ()
   from /lib/x86_64-linux-gnu/libQt5Core.so.5
#12 0x00007ffff7a7ea55 in QApplicationPrivate::notify_helper(QObject*, QEvent*) () from /lib/x86_64-linux-gnu/libQt5Widgets.so.5
#13 0x00007ffff7a88b93 in QApplication::notify(QObject*, QEvent*) () from /lib/x86_64-linux-gnu/libQt5Widgets.so.5
#14 0x00007ffff6ea593a in QCoreApplication::notifyInternal2(QObject*, QEvent*) () from /lib/x86_64-linux-gnu/libQt5Core.so.5
#15 0x00007ffff7add94a in ?? () from /lib/x86_64-linux-gnu/libQt5Widgets.so.5
#16 0x00007ffff7ae01ec in ?? () from /lib/x86_64-linux-gnu/libQt5Widgets.so.5
#17 0x00007ffff7a7ea66 in QApplicationPrivate::notify_helper(QObject*, QEvent*) () from /lib/x86_64-linux-gnu/libQt5Widgets.so.5
#18 0x00007ffff7a880f0 in QApplication::notify(QObject*, QEvent*) () from /lib/x86_64-linux-gnu/libQt5Widgets.so.5
#19 0x00007ffff6ea593a in QCoreApplication::notifyInternal2(QObject*, QEvent*) () from /lib/x86_64-linux-gnu/libQt5Core.so.5
#20 0x00007ffff728e665 in QGuiApplicationPrivate::processMouseEvent(QWindowSystemInterfacePrivate::MouseEvent*) ()
   from /lib/x86_64-linux-gnu/libQt5Gui.so.5
#21 0x00007ffff729010b in QGuiApplicationPrivate::processWindowSystemEvent(QWindowSystemInterfacePrivate::WindowSystemEvent*) ()
   from /lib/x86_64-linux-gnu/libQt5Gui.so.5
#22 0x00007ffff726a35b in QWindowSystemInterface::sendWindowSystemEvents(QFlags<QEventLoop::ProcessEventsFlag>) ()
   from /lib/x86_64-linux-gnu/libQt5Gui.so.5
#23 0x00007ffff38e232e in ?? () from /lib/x86_64-linux-gnu/libQt5XcbQpa.so.5
#24 0x00007ffff5e7d17d in g_main_context_dispatch () from /lib/x86_64-linux-gnu/libglib-2.0.so.0
#25 0x00007ffff5e7d400 in ?? () from /lib/x86_64-linux-gnu/libglib-2.0.so.0
#26 0x00007ffff5e7d4a3 in g_main_context_iteration () from /lib/x86_64-linux-gnu/libglib-2.0.so.0
#27 0x00007ffff6efd565 in QEventDispatcherGlib::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) ()
   from /lib/x86_64-linux-gnu/libQt5Core.so.5
#28 0x00007ffff6ea44db in QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) () from /lib/x86_64-linux-gnu/libQt5Core.so.5
#29 0x00007ffff6eac246 in QCoreApplication::exec() () from /lib/x86_64-linux-gnu/libQt5Core.so.5
#30 0x00005555555a06f5 in run_gui (argc=1, argv=0x7fffffffdfc8) at /home/lunarg/VulkanTools/vkconfig/main_gui.cpp:77
#31 0x00005555555a0372 in main (argc=1, argv=0x7fffffffdfc8) at /home/lunarg/VulkanTools/vkconfig/main.cpp:57
```

**Comment by christophe-lunarg** [comment]
This issue reproduce for me on macOS but not on Windows

**Comment by johnzupin** [comment]
I'm seeing an assert failing on windows for me:

![image](https://user-images.githubusercontent.com/3836693/115468031-b0c6b400-a1ef-11eb-89f8-9f73baa730e9.png)

**Comment by johnzupin** [comment]
Looks like I have Qt versions 5.14.2 and 5.15.1 on my system

**Comment by christophe-lunarg** [comment]
This issue should be fixed with This issue is fixed with https://github.com/LunarG/VulkanTools/pull/1456

I not longer reproduce this issue on macOS.

**Comment by johnzupin** [comment]
Confirming fixed as of ref: v1.2.176
